src/bylines/index.js

```diff
@@ -3,64 +3,520 @@
  */
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useRef } from 'react';
+
 /**
  * Internal dependencies
  */
 import './style.scss';
 
-const BYLINE_ID = 'newspack-byline';
+const BASE_QUERY = {
+	_fields: 'id,name',
+	context: 'view', // Allows non-admins to perform requests.
+};
+
+const AUTHORS_QUERY = {
+	who: 'authors',
+	per_page: 100,
+	...BASE_QUERY,
+};
+
+/** Close icon copied from @wordpress/icons/src/library/close.js to be used as markup */
+const close = `
+	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+		<path d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z" />
+	</svg>
+`;
+
+const CustomBylineModal = ( { children } ) => {
+	const [ isOpen, setOpen ] = useState( false );
+	const openModal = () => setOpen( true );
+	const closeModal = () => setOpen( false );
+
+	return (
+		<>
+			<Button variant="secondary" onClick={ openModal }>
+				Set Custom Byline
+			</Button>
+			{ isOpen && (
+				<Modal title="Set Custom Byline" onRequestClose={ closeModal }>
+					{ children }
+					<Button variant="secondary" onClick={ closeModal }>
+						Close
+					</Button>
+				</Modal>
+			) }
+		</>
+	);
+};
+const BylineTextarea = ( { byline, onRendered } ) => {
+	const { getEditedPostAttribute } = useSelect( select =>
+		select( 'core/editor' )
+	);
+
+	/** The custom byline stored as meta */
+	const metaByline =
+		getEditedPostAttribute( 'meta' )[ newspackBylines.metaKeyByline ] || '';
+
+	/**
+	 * Parse byline meta to convert custom tags (<Author></Author> or [Author][/Author]) to token markup.
+	 *
+	 * @see    {@link https://github.com/Automattic/newspack-plugin/tree/trunk/includes/bylines#readme|Custom Bylines}
+	 * @return {string} Parsed byline looking up for <Author id=1></Author> tags and replacing them.
+	 */
+	const bylineParser = () => {
+		const tokenMarkup = `<span id="token-$1" class="components-form-token-field__token token-inline-block author-token" data-token="$1">
+			<span class="components-form-token-field__token-text">$2</span>
+			<button
+				class="components-button components-form-token-field__remove-token token-inline-block__remove"
+				type="button"
+				data-token="$1"
+			>
+				${ close }
+			</button>
+		</span>`;
+
+		// For backwards compatibility, replace the '<Author>' tags, which were used before.
+		metaByline.replace( /<Author id=(\d*)>(\D*)<\/Author>/g, tokenMarkup );
+
+		return metaByline.replace(
+			/\[Author id=(\d*)\](\D*)\[\/Author\]/g,
+			tokenMarkup
+		);
+	};
+
+	useEffect( () => {
+		onRendered();
+	}, [ onRendered ] );
+
+	useEffect( () => {
+		byline.current = bylineParser();
+	}, [] );
+
+	return (
+		<div
+			className="newspack-byline-textarea"
+			contentEditable="true"
+			dangerouslySetInnerHTML={ { __html: byline.current } }
+		/>
+	);
+};
+
+const TokenInlineBlock = ( { token, onInsert } ) => {
+	return (
+		<>
+			<span
+				className="components-form-token-field__token token-inline-block"
```

Comment:
This probably needs `contenteditable="false"` since we're not trying to allow editing text within this span.

Reply:
Nice suggestion, I didn't thought about that